New branch

ilenghel opened the pull request:
Added Pester test for New-WindowsFromGoldenImage method

Commit:
Merge pull request #7 from costingalan/new_golden_image

Sync master branch with new_golden_image

Commit:
Added KB3172729 as blacklisted update

Signed-off-by: Costin Galan <cgalan@cloudbasesolutions.com>

ilenghel committed:
Added Pester test for New-WindowsFromGoldenImage method in the Test file and created a method to pass the resourcesDir parameter to Copy-UnattendResources method

ader1990 commented on Tests/WinImageBuilder.Tests.ps1:
```diff
@@ -181,4 +181,60 @@ Describe "Test New-MaaSImage" {
+    Mock Get-VHD -Verifiable -ModuleName $moduleName {return 0}
+    
+    It "Should Be Zero" {
+     New-WindowsFromGoldenImage -WindowsImageVHDXPath "fakeVirtualDiskPath"`
```
add indentation

ader1990 commented on Tests/WinImageBuilder.Tests.ps1:
```diff
@@ -181,4 +181,60 @@ Describe "Test New-MaaSImage" {
+    }
+    
+    It "Should Not Throw" {
+     {New-WindowsFromGoldenImage -WindowsImageVHDXPath "fakeVirtualDiskPath"`
```
add indentation

ader1990 commented on WinImageBuilder.psm1:
```diff
@@ -1078,6 +1078,16 @@ function New-WindowsCloudImage {
 
+function Get-DriveLetterOfVHDX{
+    [Parameter(Mandatory=$true)]
+    param()
```
the param definition should stay inside param()

ader1990 commented on WinImageBuilder.psm1:
```diff
@@ -1078,6 +1078,16 @@ function New-WindowsCloudImage {
+    param()
+    $driveLetter
+    
+    $driveLetter = ((Get-DiskImage -ImagePath $WindowsImageVHDXPath | Get-Disk | Get-Partition | Get-Volume).DriveLetter + ":")
```
no need for an extra variable, just return the value

ader1990 commented on WinImageBuilder.psm1:
```diff
@@ -1122,19 +1132,16 @@ function New-WindowsFromGoldenImage {
                 Resize-VHD -Path $WindowsImageVHDXPath -SizeBytes $SizeBytes
             }
-
+            
```
extra space added

Commit:
Added Pester test for New-WindowsFromGoldenImage

Added Pester test for New-WindowsFromGoldenImage method and made
the driveLetterGold be equal with the output of the method
Get-DriveLetterOfVHDX in order to write the test properly.
